main.py

```diff
@@ -360,7 +360,20 @@ async def askai(self, ctx: Context) -> None:
         response_text = response.choices[0].text.lstrip("\n")
         cents_spent = (response.usage.total_tokens * (0.02 / 1000)) * 100
 
-        # TODO: handle messages > 2000 chars
+        if len(response_text) > 2000:
+            try:
+                with open(f"/tmp/{response.openai_id}.txt", "w") as f:
+                    f.write(response_text)
+            except Exception:
+                await ctx.send("Failed to write the response to file.")
+                return
+
+            response_file = discord.File(f"/tmp/{response.openai_id}", "response.txt")
+            await ctx.send(f"Spent {cents_spent:.5f}¢ ({response.usage.total_tokens} tokens) to produce result:", file=response_file)
+            with contextlib.suppress(BaseException):
+                os.remove(f"/tmp/{response.openai_id}.txt")
```

Comment:
hey hey thanks for the pr to fix my todo :D 

i think we may be able to do a bit better - do you know if it's possible to use an in-memory file-like-object like io.bytesIO with discord.File? that would be the ideal solution, as we wouldn't have to write to disk

if that's not possible, could you use a [tempfile](https://docs.python.org/3/library/tempfile.html) with a context manager? that's a bit better than manually writing to /tmp

Reply:
fixed